crates/esplora/src/blocking_ext.rs

```diff
@@ -250,7 +250,12 @@ impl EsploraExt for esplora_client::BlockingClient {
                     }
                 }
 
-                if last_index > last_active_index.map(|i| i.saturating_add(stop_gap as u32)) {
+                let last_index = last_index.expect("Must be set since handles wasn't empty.");
+                if last_index
+                    > last_active_index
+                        .unwrap_or(0)
+                        .saturating_add(stop_gap as u32)
+                {
```

Comment:
```suggestion
                let stop_index = match last_active_index {
                    Some(active_index) => (stop_gap as u32)
                        .saturating_add(1)
                        .saturating_add(active_index),
                    None => stop_gap as u32,
                };
                if last_index >= stop_index {
```

A stop gap of 0 expects a tx history for every spk. Therefore, if we have `stop_gap = 0`, `last_index = 0` and `last_active_index = None`, we expect to break here.

Reply:
See https://github.com/bitcoindevkit/bdk/pull/1225#discussion_r1402485388, fixing this uncovers a crash. I'm checking Daniela's suggestion now.